src/Microsoft.AspNetCore.Mvc.ApiExplorer/DefaultApiDescriptionProvider.cs

```diff
@@ -162,7 +162,18 @@ private IList<ApiParameterDescription> GetParameters(ApiParameterContext context
                 foreach (var actionParameter in context.ActionDescriptor.Parameters)
                 {
                     var visitor = new PseudoModelBindingVisitor(context, actionParameter);
-                    var metadata = _modelMetadataProvider.GetMetadataForType(actionParameter.ParameterType);
+
+                    ModelMetadata metadata = null;
+                    if (actionParameter is ControllerParameterDescriptor controllerParameterDescriptor &&
```

Comment:
The `HandlerParameterDescriptor` and `PageBoundPropertyDescriptor` types are not relevant here because nobody would use Razor Pages to build a Web API. But, check if the next (`BoundProperties`) loop could be simplified using `ControllerBoundPropertyDescriptor`. If yes, fix that up too.

Reply:
Ignore my comment. This doesn't help.